src/core/meshcop/joiner.cpp

```diff
@@ -148,196 +163,93 @@ void Joiner::HandleDiscoverResult(otActiveScanResult *aResult)
     {
         otLogFuncEntryMsg("aResult = NULL");
 
-        // open UDP port
-        Ip6::SockAddr sockaddr;
-        sockaddr.mPort = mJoinerUdpPort;
-        mSocket.Open(&Joiner::HandleUdpReceive, this);
-        mSocket.Bind(sockaddr);
-
         mNetif.GetMac().SetPanId(mJoinerRouterPanId);
         mNetif.GetMac().SetChannel(mJoinerRouterChannel);
-        mNetif.GetIp6Filter().AddUnsecurePort(sockaddr.mPort);
+        mNetif.GetIp6Filter().AddUnsecurePort(mSecureCoapClient.GetPort());
 
-        mNetif.GetDtls().Start(true, HandleDtlsReceive, HandleDtlsSend, this);
+        memset(&messageInfo, 0, sizeof(messageInfo));
```

Comment:
This is no longer necessary.  The `Ip6::MessageInfo` constructor includes this `memset`.